diff --git a/flight/ExtLibraries/STM32_USB-FS-Device_Lib_V4.0.0/Libraries/STM32_USB-FS-Device_Driver/inc/usb_core.h b/flight/ExtLibraries/STM32_USB-FS-Device_Lib_V4.0.0/Libraries/STM32_USB-FS-Device_Driver/inc/usb_core.h
index 44dc327..a75f702 100644
--- a/flight/ExtLibraries/STM32_USB-FS-Device_Lib_V4.0.0/Libraries/STM32_USB-FS-Device_Driver/inc/usb_core.h
+++ b/flight/ExtLibraries/STM32_USB-FS-Device_Lib_V4.0.0/Libraries/STM32_USB-FS-Device_Driver/inc/usb_core.h
@@ -48,7 +48,8 @@
 
 typedef struct OneDescriptor
 {
-  uint8_t *Descriptor;
+/* Need to carry a const here for PiOS compatability */
+  uint8_t const *Descriptor;
   uint16_t Descriptor_Size;
 }
 ONE_DESCRIPTOR, *PONE_DESCRIPTOR;
@@ -182,9 +183,9 @@
 
   RESULT  (*Class_Get_Interface_Setting)(uint8_t Interface, uint8_t AlternateSetting);
 
-  uint8_t* (*GetDeviceDescriptor)(uint16_t Length);
-  uint8_t* (*GetConfigDescriptor)(uint16_t Length);
-  uint8_t* (*GetStringDescriptor)(uint16_t Length);
+  uint8_t const * (*GetDeviceDescriptor)(uint16_t Length);
+  uint8_t const * (*GetConfigDescriptor)(uint16_t Length);
+  uint8_t const * (*GetStringDescriptor)(uint16_t Length);
 
   /* This field is not used in current library version. It is kept only for 
    compatibility with previous versions */